tft.py

```diff
@@ -0,0 +1,189 @@
+
+            champ_traits = []
+            tft_bin = BinFile(os.path.join(character_folder, name, name + ".bin"))
+            record = [x for x in tft_bin.entries if x.type == "TFTCharacterRecord"][0]
+            for trait in record.getv("mLinkedTraits"):
+                champ_traits.append([x["internal"]["name"] for x in traits if x["internal"]["hash"] == trait.hex()][0])
+
+            stats_obj = {
+                "hp": record.getv("baseHP"),
+                "mana": record.getv("primaryAbilityResource").getv("arBase"),
+                "initialMana": record.getv("mInitialMana") or 0,
+                "damage": record.getv("BaseDamage"),
+                "armor": record.getv("baseArmor"),
+                "magicResist": record.getv("baseSpellBlock"),
+                "critMultiplier": record.getv("critDamageMultiplier"),
+                "critChance": record.getv("baseCritChance"),
+                "attackSpeed": record.getv("attackSpeed"),
+                "range": record.getv("attackRange") // 180,
+            }
+
+            spell_name = record.getv("spellNames")[0]
+            if "/" in spell_name:
+                spell_name = os.path.basename(spell_name)
+            ability = [x.getv("mSpell") for x in tft_bin.entries if x.type == "SpellObject" and x.getv("mScriptName") == spell_name][0]
+
+            variables = []
+            for value in ability.getv("mDataValues"):
+                variables.append({"name": value.getv("mName"), "value": value.getv("mValues")})
+
+            ability_obj = {"name": champ.getrv("87a69a5e"), "desc": champ.getrv("bc4f18b3"), "icon": champ.getv("mPortraitIconPath"), "variables": variables}
+
+            champs.append(
+                {
+                    "id": champ_id,
+                    "name": champ.getrv("c3143d66"),
+                    "cost": champ.getv("mRarity") or 1,
+                    "icon": champ.getv("mIconPath"),
+                    "traits": champ_traits,
+                    "stats": stats_obj,
+                    "ability": ability_obj,
+                }
+            )
+
+        return champs
+
+    def parse_traits(self, map22):
+        trait_collection = [x for x in map22.entries if x.type == "TftTraitData"]
+        traits = []
+
+        for trait in trait_collection:
+            if "Template" in trait.getv("mName"):
+                continue
+
+            effects = []
+            for trait_set in trait.getv("mTraitSets"):
+                variables = []
+                for effect in trait_set.getv("effectAmounts") if trait_set.get("effectAmounts") else []:
+                    variables.append({"name": effect.getv("name"), "value": effect.getv("value")})
+
+                effects.append({"minUnits": trait_set.getv("mMinUnits"), "maxUnits": trait_set.getv("mMaxUnits"), "variables": variables})
+
+            traits.append(
+                {
+                    "internal": {"hash": trait.path.hex(), "name": trait.getv("mName")},
```

Comment:
You should use `trait.path` or `trait.path.h` directly instead of the hex string.